test/compflags/checks/checkModByZero.chpl

```diff
@@ -0,0 +1,62 @@
+/* Tests for modulus div-by-zero error */
+
+config param testParamCases = 0;
+config const testExecCases = 0;
+
+var num = 10,
+    unum = 10: uint,
+    denom = 0,
+    udenom = 0: uint;
+
+record R {
+  var x = 10;
+}
+
+/* Define this so that %= is automatically defined */
+proc %(lhs: R, rhs: R) return new R(lhs.x % rhs.x);
+
+var rec1 = new R(1),
+    rec0 = new R(0);
+
+select testParamCases {
+  //proc %(param a: int(?w), param b: int(w)) param
+  when 1 do
+    writeln(10 % 0);
+
+  //proc %(param a: uint(?w), param b: uint(w)) param
+  when 2 do
+    writeln(10 % 0);
+
+  //proc %(a: uint(64), param b: uint(64))
+  when 3 do
+    writeln(unum % (0:uint));
```

Comment:
I'd guess that the cast here isn't necessary (and the test might be slightly more interesting without it since it's what users are more likely to write).  If it is necessary, I'd be curious to know why.